ui/app/src/modules/Preview/PreviewConcierge.tsx

```diff
@@ -469,23 +469,25 @@ export function PreviewConcierge(props: any) {
         case DESKTOP_ASSET_DROP: {
           enqueueSnackbar('Asset upload started.');
           hostToHost$.next({ type: DESKTOP_ASSET_UPLOAD_STARTED, payload });
-          const uppySubscription = uploadDataUrl(
+          const uppySubscription = createFileUpload(
             site,
             pluckProps(payload, 'name', 'type', 'dataUrl'),
             `/static-assets/images/${payload.record.modelId}`,
             xsrfArgument
           ).subscribe(
             ({ payload: { progress } }) => {
-              const percentage = Math.floor(
-                parseInt(((progress.bytesUploaded / progress.bytesTotal) * 100).toFixed(2))
-              );
-              hostToGuest$.next({
-                type: DESKTOP_ASSET_UPLOAD_PROGRESS,
-                payload: {
-                  record: payload.record,
-                  percentage
-                }
-              });
+              if (progress) {
```

Comment:
Why wouldn't there be progress? What's the need for this?

This will skip a possible 0%, which is a valid progress number.

Reply:
Changed to validate type of action instead of progress